src/catchup/CatchupWork.cpp

```diff
@@ -106,12 +107,27 @@ CatchupWork::alreadyHaveBucketsHistoryArchiveState(uint32_t atCheckpoint) const
 WorkSeqPtr
 CatchupWork::downloadApplyBuckets()
 {
+    // If first ledger we're going to replay is of older version (before shadow
+    // removal), it'll perform old-style merges
+    auto ledgerVersion = mVerifiedLedgerRangeStart.header.ledgerVersion;
```

Comment:
why not just always wait for merges regardless of protocol version? The danger of blocking the main thread exists also in the current version of the protocol, right?